src/App.tsx

```diff
@@ -1,7 +1,37 @@
-import React from 'react';
+import { Component } from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  pressedKey: string,
+};
+
+export class App extends Component<{}, State> {
+  state = {
+    pressedKey: 'Nothing was pressed yet',
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', this.keypress);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', this.keypress);
+  }
+
+  keypress = (event: KeyboardEvent) => {
+    if (event.code === 'Space') {
+      this.setState({ pressedKey: `The last pressed key is [${event.code}]` });
+    } else {
+      this.setState({ pressedKey: `The last pressed key is [${event.key}]` });
+    }
```

Comment:
The only difference in your statements is `event.code` and `event.key`, the rest is the same. 
How can you make your code DRY and not duplicate the rest info?

And use conditional operators `?` and `:`